packages/plugin-pnp/sources/PnpLinker.ts

```diff
@@ -1,13 +1,12 @@
-import {Linker, LinkOptions, MinimalLinkOptions, Manifest, MessageName, DependencyMeta} from '@yarnpkg/core';
-import {FetchResult, Ident, Locator, Package, BuildDirective, BuildType}                from '@yarnpkg/core';
-import {miscUtils, structUtils, formatUtils}                                            from '@yarnpkg/core';
-import {CwdFS, FakeFS, PortablePath, npath, ppath, xfs, Filename}                       from '@yarnpkg/fslib';
-import {generateInlinedScript, generateSplitScript, PnpSettings}                        from '@yarnpkg/pnp';
-import {UsageError}                                                                     from 'clipanion';
+import {miscUtils, structUtils, formatUtils, Descriptor, LocatorHash}                                        from '@yarnpkg/core';
+import {FetchResult, Locator, Package}                                                                       from '@yarnpkg/core';
+import {Linker, LinkOptions, MinimalLinkOptions, Manifest, MessageName, DependencyMeta, LinkType, Installer} from '@yarnpkg/core';
+import {CwdFS, FakeFS, PortablePath, npath, ppath, xfs, Filename}                                            from '@yarnpkg/fslib';
+import {generateInlinedScript, generateSplitScript, PackageRegistry, PnpSettings}                            from '@yarnpkg/pnp';
+import {UsageError}                                                                                          from 'clipanion';
 
-import {AbstractPnpInstaller}                                                           from './AbstractPnpInstaller';
-import {getPnpPath}                                                                     from './index';
-import * as pnpUtils                                                                    from './pnpUtils';
+import {getPnpPath, javascriptUtils}                                                                         from './index';
```

Comment:
Impressive this doesn't/didn't throw at runtime because this seems like a circular import